lib/puppet/configurer.rb

```diff
@@ -5,6 +5,17 @@
 #require 'puppet/parser/script_compiler'
 require_relative '../puppet/pops/evaluator/deferred_resolver'
 
+# max fact name length in bytes
+MAX_FACT_NAME_LENGTH = 2713
+# max fact value length in bytes
+MAX_FACT_VALUE_LENGTH = 512
+# max fact top level facts
+MAX_TOP_LEVEL_FACTS = 500
+# max fact top level facts
```

Comment:
Looks like copy/paste error? I’m assuming it’s the maximum number of total facts?